### DISCUSSION TITLE
Why do people want issues to be assigned to them?

### DISCUSSION THREAD

**Discussion by tianyizheng02** [TheAlgorithms/Python#8866] [General]
I sometimes see users commenting on this repo and asking to be assigned to an issue, and in particular this happens constantly during Hacktoberfest. Why not just contribute? Is there any particular reason why one would need an issue to be assigned to them?

**Comment by CaedenPH** [comment]
> I sometimes see users commenting on this repo and asking to be assigned to an issue, and in particular this happens constantly during Hacktoberfest. Why not just contribute? Is there any particular reason why one would need an issue to be assigned to them?

@tianyizheng02 
Typically in other repos you can only submit a pr for an issue that is assigned to you, to stop multiple people working on the same problem and allow the people who the issue is assigned to to gain experience. 

For Hacktoberfest, getting issues assigned to themselves is an easy way to get ideas and an easy way to obtain essentially a "permit" to implement those ideas. This means that people can just write the code without having to think of any idea and get the idea assigned to them (perhaps the more laborious aspects of submitting a pr) 

Most of the comments where people ask to be assigned are bots anyway, to make it easier for the owner of the bot to submit the prs they need.

**Reply by tianyizheng02** [reply]
That makes sense, thank you. In any case, I think we should explicitly state somewhere (like in the contributing guidelines, for example) that we don't assign issues in this repo.

**Reply by CaedenPH** [reply]
> That makes sense, thank you. In any case, I think we should explicitly state somewhere (like in the contributing guidelines, for example) that we don't assign issues in this repo.

Yeah that sounds like a good thing to add to the CONTRIBUTING. Feel free to make a pull request. If you don't want to, I'm happy to.

If you don't have any more questions mark my answer as correct or close the discussion so people can filter via unanswered discussions
